DEVELOPMENT_GUIDE.md

```diff
@@ -21,93 +17,76 @@ This will start a new Gitpod workspace, and immediately kick off a build of the
+| Activate the virtual environment | `source .venv/bin/activate` | `pyenv activate samcli38` |
 
-However, IDEs such PyChaim (using FileWatcher) will have a hard time invoking `/Users/<username>/.pyenv/shims/black` 
-and this will happen:
 
-```
-pyenv: black: command not found
 
-The `black' command exists in these Python versions:
-  3.7.2/envs/samcli37
-  samcli37
-``` 
+**For convenience, below we assume the virtual environment is called `samcli38`.**
 
-A simple workaround is to use `/Users/<username>/.pyenv/versions/samcli37/bin/black` 
-instead of `/Users/<username>/.pyenv/shims/black`.
+### 2. Initialize dependencies and create `samdev` available in `$PATH`
 
-#### Pre-commit
-If you don't wish to manually run black on each pr or install black manually, we have integrated black into git hooks through [pre-commit](https://pre-commit.com/).
-After installing pre-commit, run `pre-commit install` in the root of the project. This will install black for you and run the black formatting on
-commit.
+Clone the AWS SAM CLI repository to your local machine if you haven't done that yet.
 
-### 3. Activate Virtualenv
-
-Virtualenv allows you to install required libraries outside of the
-Python installation. A good practice is to setup a different virtualenv
-for each project. [pyenv](https://github.com/pyenv/pyenv) comes with a
-handy plugin that can create virtualenv.
+```sh
+# Using SSH
+$ git clone git@github.com:aws/aws-sam-cli.git
+# Using HTTPS
+$ git clone https://github.com/aws/aws-sam-cli.git
+```
 
-Depending on the python version, the following commands would change to
-be the appropriate python version.
+(make sure you have virtual environment activated)
 
-1.  `pyenv virtualenv 3.7.2 samcli37`
-2.  `pyenv activate samcli37` for Python3.7
+```sh
+(samcli38) $ cd aws-sam-cli
+(samcli38) $ make init  # this will put a file `samdev` available in $PATH
+```
 
-### 4. Install dev version of SAM CLI
+Now you can verify whether the dev AWS SAM CLI is available:
 
-We will install a development version of SAM CLI from source into the
-virtualenv for you to try out the CLI as you make changes. We will
-install in a command called `samdev` to keep it separate from a global
-SAM CLI installation, if any.
+```sh
+(samcli38) $ which samdev
+/<some-path>/bin/samdev
+(samcli38) $ samdev --version
+SAM CLI, version x.xx.x
+```
 
-1.  Activate Virtualenv: `pyenv activate samcli37`
-2.  Install dev CLI: `make init`
-3.  Make sure installation succeeded: `which samdev`
+#### Try out to make change to AWS SAM CLI (Optional)
+
+```sh
+# Change the AWS SAM CLI version to 123.456.789
+(samcli38) $ echo '__version__ = "123.456.789"' >> samcli/__init__.py
+(samcli38) $ samdev --version
+SAM CLI, version 123.456.789
+# Dismiss the change
+(samcli38) $ git checkout -- samcli/__init__.py
+(samcli38) $ samdev --version  # the original version should be printed
+SAM CLI, version x.xx.x
```

Comment:
I'd get rid of this. Unrelated to making changes and it's not a Git tutorial (nor is it changing a system-wide `sam`).

Reply:
+1

Reply:
removed